qa/rpc-tests/disablehot.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python3
+# Copyright (c) 2017 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import *
+
+class DisableHotKeysTest(BitcoinTestFramework):
+
+    def __init__(self):
+        super().__init__()
+        self.setup_clean_chain = True
+        self.num_nodes = 2
+
+    def setup_network(self, split=False):
+        self.nodes = start_nodes(self.num_nodes, self.options.tmpdir, [[], ['-disablehot']])
+
+        connect_nodes_bi(self.nodes,0,1)
+
+        self.is_network_split=False
+        self.sync_all()
+
+    def run_test(self):
+        assert_equal(self.nodes[1].getwalletinfo()['keypoolsize'], 0)
+
+        print("Mining blocks...")
+        self.sync_all()
+        self.nodes[0].generate(101)
+        self.sync_all()
+
+        n0addrR = self.nodes[0].getnewaddress()
+        n0pubkR = self.nodes[0].validateaddress(n0addrR)['pubkey']
+        n0addrC = self.nodes[0].getnewaddress()
+        n0pubkC = self.nodes[0].validateaddress(n0addrC)['pubkey']
+        n0addr  = self.nodes[0].getnewaddress()
+        self.nodes[0].sendtoaddress(n0addrR, 10);
+        self.nodes[0].generate(6)
+        self.sync_all()
+
+        try:
+            self.nodes[1].getnewaddress()
+            raise AssertionError("Wallet can derive private keys while -disablehot is enabled")
+        except JSONRPCException as e:
+            assert('Hot key' in e.error['message'])
+
+        try:
+            self.nodes[1].getrawchangeaddress()
+            raise AssertionError("Wallet can derive private keys while -disablehot is enabled")
+        except JSONRPCException as e:
+            assert('Hot key' in e.error['message'])
+
+        self.nodes[1].importpubkey(n0pubkR) #TODO switch to importmulti
+        self.nodes[1].importpubkey(n0pubkC)
+        assert_equal(self.nodes[1].getbalance("*", 1, True), 10)
+        rawtx = self.nodes[1].createrawtransaction([], {n0addr: 1.0})
+        frawtx= self.nodes[1].fundrawtransaction(rawtx, {"changeAddress": n0addrC, "includeWatching": True})
+
+        # check if we can re-enable private keys
+        assert_equal(self.nodes[1].signrawtransaction(frawtx['hex'])['complete'], False)
+        print("restarting node with disablehot=0 (try to re-enable private keys which must not be possible)")
+        stop_nodes(self.nodes)
+        try:
+            self.nodes = start_nodes(self.num_nodes, self.options.tmpdir, [[], ['-disablehot=0']])
```

Comment:
@MarcoFalke: any idea how I can make this pass? Right now, staring bitcoind in this case with this setup will cause bitcoind to write to stderr (and halt) which then result in pass=false in `rpc-tests.py`. But it's actually the correct behaviour.